### PR TITLE
feat(vpn): huaweicloud_vpn_gateway support extra field

### PR DESCRIPTION
**Pull request by dangyy0626**
**What this PR does / why we need it**:
support extra volume: "ha_mode","eip1","eip2"
master_eip can be replace by eip1
slave_eip can be replace by eip2

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccGateway_basic" 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_basic -timeout 360m -parallel 4 
=== RUN   TestAccGateway_basic 
=== PAUSE TestAccGateway_basic
=== CONT  TestAccGateway_basic
--- PASS: TestAccGateway_basic (344.90s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       344.933s
```

deprecated volume is still supported
```
make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccGateway_deprecated" 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_deprecated -timeout 360m -parallel 4 
=== RUN   TestAccGateway_deprecated 
=== PAUSE TestAccGateway_deprecated
=== CONT  TestAccGateway_deprecated
--- PASS: TestAccGateway_deprecated (405.48s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       405.514s

```

```
make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccGateway_activeStandbyHAMode" 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_activeStandbyHAMode -timeout 360m -parallel 4 
=== RUN   TestAccGateway_activeStandbyHAMode 
=== PAUSE TestAccGateway_activeStandbyHAMode
=== CONT  TestAccGateway_activeStandbyHAMode
--- PASS: TestAccGateway_activeStandbyHAMode (272.09s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       272.137s
```

### CODE REVIEW

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go [hunk 1]
```diff
@@ -121,21 +121,51 @@ func ResourceGateway() *schema.Resource {
 				Computed:    true,
 				Description: `The enterprise router ID to attach with to VPN gateway.`,
 			},
+			"ha_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				Description:  `The HA mode of the VPN gateway.`,
+				ValidateFunc: validation.StringInSlice([]string{"active-active", "active-standby"}, false),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go]: the ha_mode is fixed to "active-standby" when "master_eip" and "slave_eip" are specified
```suggestion
				ValidateFunc: validation.StringInSlice([]string{"active-active", "active-standby"}, false),
				ConflictsWith: []string{"master_eip", "slave_eip"},
```

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go [hunk 2]
```diff
@@ -121,21 +121,51 @@ func ResourceGateway() *schema.Resource {
...
-				Optional: true,
-				ForceNew: true,
-				Computed: true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        GatewayEipSchema(),
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: utils.SchemaDesc("", utils.SchemaDescInput{Internal: true}),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go]: better to add a RequiredWith here

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go [hunk 3]
```diff
@@ -317,83 +347,82 @@ func resourceGatewayCreate(ctx context.Context, d *schema.ResourceData, meta int
...
 	return bodyParams
 }
 
-func buildCreateGatewayVpnGatewayChildBody(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+func buildCreateGatewayVpnGatewayChildBody(d *schema.ResourceData, cfg *config.Config) map[string]interface{} {
+	haMode := d.Get("ha_mode").(string)
+	masterEIP := buildCreateGatewayEIPChildBody(d, "master_eip")
+	slaveEIP := buildCreateGatewayEIPChildBody(d, "slave_eip")
+
+	// 保留之前的默认创建ha_mode类型，master_eip和slave_eip下线后一起删除
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go]: use English please

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go [hunk 4]
```diff
@@ -603,7 +615,7 @@ func resourceGatewayUpdate(ctx context.Context, d *schema.ResourceData, meta int
 	return resourceGatewayRead(ctx, d, meta)
 }
 
-func buildUpdateGatewayBodyParams(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+func buildUpdateGatewayBodyParams(d *schema.ResourceData, _ config.Config) map[string]interface{} {
```
Comment (ShiChangkuo) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go]: how about delete the last param instead of using `_`?

**Pull request merged**